packages/controls/src/widget_image.ts

```diff
@@ -44,11 +44,20 @@ class ImageView extends DOMWidgetView {
          * Called when the model is changed.  The model may have been
          * changed by another view or by a state update from the back-end.
          */
-        let blob = new Blob([this.model.get('value')], {type: `image/${this.model.get('format')}`});
-        let url = URL.createObjectURL(blob);
-        let oldurl = this.el.src;
+        let url;
+        let format = this.model.get('format');
+        let value = this.model.get('value');
+        if (format != 'url') {
+            var blob = new Blob([value], {type: `image/${this.model.get('format')}`});
+            url = URL.createObjectURL(blob);
+        } else {
+            url = String.fromCharCode.apply(null, new Uint8Array(value.buffer));
```

Comment:
Does this work for utf8 urls? See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/fromCharCode#Getting_it_to_work_with_higher_values.

It's probably better to use the TextDecoder api (and shim for IE?): https://developer.mozilla.org/en-US/docs/Web/API/TextDecoder

Reply:
In fact, apparently the notebook itself uses TextDecoder to get the JSON parts of a binary message: https://github.com/jupyter/notebook/blob/179bb24fbf79d153812858126127a91431da3319/notebook/static/services/kernels/serialize.js#L20

Reply:
Weird that it is used in the notebook, especially since it is not supported in IE 11 and in development in Edge: https://developer.microsoft.com/en-us/microsoft-edge/platform/status/encodingstandard/

Reply:
Looks like the notebook already includes a polyfill: https://github.com/jupyter/notebook/issues/3457

I think we should assume the TextDecoder api works on the page. The notebook has a polyfill, and lab doesn't support IE/Edge. People embedding widgets, if the widgets use binary messages, will need to put it on the page.